articles/best-practices/custom-db-connections/environment.md

```diff
@@ -0,0 +1,63 @@
+
+The `configuration` object can also be used to support whatever [Software Development Life Cycle (SDLC)](/dev-lifecycle/setting-up-env) best practice strategies you employ by allowing you to define variables that have tenant specific values. This mitigates hard code values in an action script which may change depending upon which tenant is executing it.
+
+## `global` object
+
+Auth0 serverless Webtask containers are provisioned from a pool that's associated with each Auth0 tenant. Each container instance makes available a global object which can be accessed across all action scripts that execute within it (the container instance). The global object acts as a global variable that’s unique to the container, and that can be used to define information - or even functions - that can be used across all action scripts that run in it (the container instance).
+
+This means that the global object can be used to cache expensive resources, as long as those resources are not user-specific. For example, an Access Token for a third-party (e.g., logging) API that provides non user-specific functionality could be stored. Or it could be used to store an Access Token to your own non user-specific API defined in Auth0 and obtained via use of Client Credentials flow.
+
+::: warning
+An action script may execute in any of the container instances already running, or in a newly created container instance (which may subsequently be added to the pool). There is no container affinity *per-se* for action script execution in Auth0. This means that you should avoid storing any user-specific information in the global object, and should always ensure that any declaration made within the global object provides for initialization too.
+:::
+
+Each time a Webtask container is recycled, or for each instantiation of a new Webtask container, the global object it defines is reset. Thus any declaration of assignment within the global object associated with a container should also include provision for initialization too. To provide performance flexibility, serverless Webtask containers are provisioned in Auth0 on an ad-hoc basis and are also subject to various re-cycle policies. In general, we recommend that you do not consider the life of a global object to be anything more than 20 minutes.
+
+## Custom database connection environment checklist
+
+* Make sure that your database has the appropriate fields to store user profiles attributes, such as **id**, **nickname**, **email**, and **password**. See [Normalized User Profile](/users/normalized) for details on Auth0's user profile schema and the expected fields. Also, see [Update User Profile Using Your Database](/users/guides/update-user-profiles-using-your-database) for more information.
```

Comment:
This is somewhat ambiguous. Custom (legacy) identity storage need only provide storage for **email**, **password**, **name** (optional) and a flag to indicate whether or not the email has been verified. Storage for Normalized User Profile attributes doesn't need to be provided - these are typically store in the Auth0 User Account database against the user's profile. I'm not really sure how we should articulate this - @mostekcm what do you think?